Core/src/org/sleuthkit/autopsy/contentviewers/MediaPlayerPanel.java

```diff
@@ -0,0 +1,681 @@
+    }
+
+    private boolean initGst() {
+        try {
+            logger.log(Level.INFO, "Initializing gstreamer for video/audio viewing"); //NON-NLS
+            Gst.init();
+            gstInited = true;
+        } catch (GstException e) {
+            gstInited = false;
+            logger.log(Level.SEVERE, "Error initializing gstreamer for audio/video viewing and frame extraction capabilities", e); //NON-NLS
+            MessageNotifyUtil.Notify.error(
+                    NbBundle.getMessage(this.getClass(), "GstVideoPanel.initGst.gstException.msg"),
+                    e.getMessage());
+            return false;
+        } catch (UnsatisfiedLinkError | NoClassDefFoundError | Exception e) {
+            gstInited = false;
+            logger.log(Level.SEVERE, "Error initializing gstreamer for audio/video viewing and extraction capabilities", e); //NON-NLS
+            MessageNotifyUtil.Notify.error(
+                    NbBundle.getMessage(this.getClass(), "GstVideoPanel.initGst.otherException.msg"),
+                    e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Initialize all the necessary variables to play an audio/video file.
+     *
+     * @param file Media file to play.
+     * @param dims Dimension of the parent window.
+     */
+    @NbBundle.Messages ({"GstVideoPanel.noOpenCase.errMsg=No open case available."})
+    void loadFile(final AbstractFile file, final Dimension dims) {
+        EventQueue.invokeLater(() -> {
+            reset();
+            infoLabel.setText("");
+            currentFile = file;
+            final boolean deleted = file.isDirNameFlagSet(TskData.TSK_FS_NAME_FLAG_ENUM.UNALLOC);
+            if (deleted) {
+                infoLabel.setText(NbBundle.getMessage(this.getClass(), "GstVideoPanel.setupVideo.infoLabel.text"));
+                videoPanel.removeAll();
+                pauseButton.setEnabled(false);
+                progressSlider.setEnabled(false);
+                return;
+            }
+
+            java.io.File ioFile;
+            try {
+                ioFile = VideoUtils.getVideoFileInTempDir(file);
+            } catch (NoCurrentCaseException ex) {
+                logger.log(Level.SEVERE, "Exception while getting open case.", ex); //NON-NLS
+                infoLabel.setText(Bundle.GstVideoPanel_noOpenCase_errMsg());
+                pauseButton.setEnabled(false);
+                progressSlider.setEnabled(false);
+
+                return;
+            }
+
+            String path = "";
+            try {
+                path = file.getUniquePath();
+            } catch (TskCoreException ex) {
+                logger.log(Level.SEVERE, "Cannot get unique path of video file"); //NON-NLS
+            }
+            infoLabel.setText(path);
+            infoLabel.setToolTipText(path);
+            pauseButton.setEnabled(true);
+            progressSlider.setEnabled(true);
+            timer = new Timer(50, e -> {
```

Comment:
What is "50" input argument that is getting passed in as input? I think it should be a defined constant instead.